app/services/calculators/state_pension_date_calculator.rb

```diff
@@ -0,0 +1,343 @@
+module Calculators
+  class StatePensionDateCalculator
+    class << self
+      AGE = "age".freeze
+      FIXED = "fixed".freeze
+
+      # Calculates the date on which a UK citizen becomes eligible for their State Pension.
+      def state_pension_date(date_of_birth:)
+        pension_rule = PENSION_TABLE.detect do |rule|
+          start_date = Date.parse(rule[:periodStart])
+          end_date = Date.parse(rule.fetch(:periodEnd, "2525-01-01"))
```

Comment:
I think we should have used a named constant here (FAR_FLUNG_FUTURE_DATE?) and set it to 2999-12-31